QUARKUS-6264.md

```diff
@@ -0,0 +1,33 @@
+# QUARKUS-6264 - Bump kafka-clients from 3.9.0 to 4.0.0
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-6264
+
+Upstream PRs:
+    - https://github.com/quarkusio/quarkus/pull/47108
+
+This change bumps kafka-client dependencies to new mayor version 4.0.0
+
+## Getting familiar with the issue
+In quarkus ecosystem kafka-clients is available indirectly via `quarkus-messaging-kafka` or`quarkus-kafka-client`.
+Quarkus' changelog and migration guide does not state any new features or notable changes related to this bump.
+Also, kafka's [release announcement](https://kafka.apache.org/blog#apache_kafka_400_release_announcement),
+[release notes](https://archive.apache.org/dist/kafka/4.0.0/RELEASE_NOTES.html) and
+[upgrade guide](https://kafka.apache.org/40/documentation.html#upgrade) doesn't state changes significant for us.
+For kafka-clients There are a lot of internal implementation changes, requirement for java 11, but very little changes that would be notable for users in Quarkus.
+
+## Scope of testing
+Main goal of this testing is to check for regressions related with this dependency update.
+Since no notable changes are expected no new tests should be required.
+
+Updating kafka to new mayor version should also be done in `quarkus-test-framework`, by bumping kafka broker.
+This might lead to problems requiring changes in FW/TS to work correctly or expose problems with compatibility.
+
+## Existing test coverage.
+Our test suite already has significant test coverage in several `messaging/kafka-*` modules.
```

Comment:
In OpenShift testing - do we depend on Strimzi or Streams for Apache Kafka product?

Reply:
https://github.com/quarkus-qe/quarkus-test-suite/blob/main/messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/OperatorOpenShiftAmqStreamsKafkaStreamIT.java
https://github.com/quarkus-qe/quarkus-test-suite/blob/main/pom.xml#L807

Product is also test, you will need to look into TS adjustments too.